pydra/engine/core.py

```diff
@@ -1227,21 +1232,31 @@ def create_dotfile(self, type="simple", export=None, name=None):
                 formatted_dot.append(self.graph.export_graph(dotfile=dotfile, ext=ext))
             return dotfile, formatted_dot
 
-    def _connect_and_propagate_to_tasks(self):
+    def _connect_and_propagate_to_tasks(
+        self,
+        *,
+        propagate_rerun=False,
+        override_task_caches=False,
+    ):
         """
         Visit each node in the graph and create the connections.
         Additionally checks if all tasks should be rerun.
         """
         for task in self.graph.nodes:
+            self.create_connections(task)
             # if workflow has task_rerun=True and propagate_rerun=True,
             # it should be passed to the tasks
-            if self.task_rerun and self.propagate_rerun:
-                task.task_rerun = self.task_rerun
+            if propagate_rerun:
+                task.task_rerun = True
```

Comment:
I'm wondering if this should be always `True`, even if `self.task_rerun` is `False`?

Reply:
it should never be False, since we call this as:

```python
        self._connect_and_propagate_to_tasks(
            propagate_rerun=self.task_rerun and self.propagate_rerun
        )
```